frontstage/controllers/case_controller.py

```diff
@@ -0,0 +1,160 @@
+    if case_events:
+        url = f'{url}?caseevents=true'
+    response = request_handler('GET', url, auth=app.config['BASIC_AUTH'])
+
+    if response.status_code != 200:
+        raise ApiError(url=url, status_code=response.status_code, description='Failed to retrieve case',
+                       party_id=party_id)
+
+    logger.debug('Successfully retrieved case', party_id=party_id)
+    return json.loads(response.text)
+
+
+def get_case_by_enrolment_code(enrolment_code):
+    logger.debug('Retrieving case', enrolment_code=enrolment_code)
+    url = f"{app.config['RM_CASE_SERVICE']}/cases/iac/{enrolment_code}"
+    response = request_handler('GET', url, auth=app.config['BASIC_AUTH'])
+
+    if response.status_code != 200:
+        raise ApiError(url=url, status_code=response.status_code, description='Failed to retrieve case',
+                       enrolment_code=enrolment_code)
+
+    logger.debug('Successfully retrieved case', enrolment_code=enrolment_code)
+    return json.loads(response.text)
+
+
+def get_case_categories():
+    logger.debug('Retrieving case categories')
+    url = f"{app.config['RM_CASE_SERVICE']}/categories"
+    response = request_handler('GET', url, auth=app.config['BASIC_AUTH'])
+
+    if response.status_code != 200:
+        raise ApiError(url=url, status_code=response.status_code, description='Failed to retrieve case categories')
+
+    logger.debug('Successfully retrieved case categories')
+    return json.loads(response.text)
+
+
+def validate_case_category(category):
+    logger.debug('Validating case category', category=category)
+    categories = get_case_categories()
+    category_names = [cat['name'] for cat in categories]
+    if category not in category_names:
+        raise InvalidCaseCategory(category)
+
+
+def post_case_event(case_id, party_id, category, description):
+    logger.debug('Posting case event', case_id=case_id)
+    validate_case_category(category)
+    url = f"{app.config['RM_CASE_SERVICE']}/cases/{case_id}/events"
+    message = {
+        'description': description,
+        'category': category,
+        'partyId': party_id,
+        'createdBy': 'RAS_FRONTSTAGE_API'
+    }
+    response = request_handler('POST', url, auth=app.config['BASIC_AUTH'], json=message)
+
+    if response.status_code != 201:
+        raise ApiError(url=url, status_code=response.status_code, description='Failed to post to case service',
+                       case_id=case_id)
+    logger.debug('Successfully posted case event', case_id=case_id)
+
+
+def check_case_permissions(party_id, case_party_id, case_id=None):
+    logger.debug('Party requesting access to case', party_id=party_id, case_id=case_id, case_party_id=case_party_id)
+    if party_id != case_party_id:
+        raise NoSurveyPermission(party_id, case_id, case_party_id)
+
+    logger.debug('Party has permission to access case', party_id=party_id, case_id=case_id, case_party_id=case_party_id)
+
+
+def build_full_case_data(case):
```

Comment:
It's probably best to leave this piece of functionality in frontstage-api for now.    To make the PR easier to read, and to reduce the risk, it should only be the code relating to the case service that should be done here.
The config changes above should be reasonably small (they should only be urls for the collection instrument service and nothing else)

Reply:
@insacuri the whole module was copied over to support the required functionality. If this is something (among others) that can be left behind, then I'll do so.